File: app/src/main/java/com/medziku/plantwatering/data/parts/SunlightSensorImpl.java
package com.medziku.plantwatering.data.parts;

import android.os.Parcel;
import android.os.Parcelable;

import com.medziku.plantwatering.station.parts.SunlightSensor;


public class SunlightSensorImpl extends SunlightSensor implements Parcelable {

    public SunlightSensorImpl(String name, float sunlightLevel) {
        super(name, sunlightLevel);
    }

    private SunlightSensorImpl(Parcel in) {
        super(in.readString(), in.readFloat());
    }

    public static final Creator<SunlightSensorImpl> CREATOR = new Creator<SunlightSensorImpl>() {
        @Override
        public SunlightSensorImpl createFromParcel(Parcel in) {
            return new SunlightSensorImpl(in);
        }

        @Override
        public SunlightSensorImpl[] newArray(int size) {
            return new SunlightSensorImpl[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(getName());
        dest.writeFloat(getSunlightLevel());
    }
}

File: android-app/src/main/java/com/medziku/plantwatering/data/parts/SunlightSensorImpl.kt
package com.medziku.plantwatering.data.parts

import android.os.Parcel
import android.os.Parcelable

import com.medziku.plantwatering.station.parts.SunlightSensor


class SunlightSensorImpl : SunlightSensor, Parcelable {

    constructor(name: String, sunlightLevel: Float) : super(name, sunlightLevel)

    private constructor(`in`: Parcel) : super(`in`.readString(), `in`.readFloat())

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeFloat(sunlightLevel)
    }

    companion object {

        val CREATOR: Parcelable.Creator<SunlightSensorImpl> = object : Parcelable.Creator<SunlightSensorImpl> {
            override fun createFromParcel(`in`: Parcel): SunlightSensorImpl {
                return SunlightSensorImpl(`in`)
            }

            override fun newArray(size: Int): Array<SunlightSensorImpl?> {
                return arrayOfNulls(size)
            }
        }
    }
}
